cub/cub/device/dispatch/kernels/kernel_segmented_scan.cuh

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+
+#include <cub/config.cuh>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cub/agent/agent_segmented_scan.cuh>
+#include <cub/util_macro.cuh>
+#include <cub/util_type.cuh>
+
+CUB_NAMESPACE_BEGIN
+
+namespace detail::segmented_scan
+{
+template <typename ChainedPolicyT,
+          typename InputIteratorT,
+          typename OutputIteratorT,
+          typename BeginOffsetIteratorInputT,
+          typename EndOffsetIteratorInputT,
+          typename BeginOffsetIteratorOutputT,
+          typename OffsetT,
+          typename ScanOpT,
+          typename InitValueT,
+          typename AccumT,
+          bool ForceInclusive,
+          typename ActualInitValueT = typename InitValueT::value_type>
+__launch_bounds__(int(ChainedPolicyT::ActivePolicy::SegmentedScanPolicyT::BLOCK_THREADS))
+  CUB_DETAIL_KERNEL_ATTRIBUTES void DeviceSegmentedScanKernel(
+    InputIteratorT d_in,
+    OutputIteratorT d_out,
+    BeginOffsetIteratorInputT begin_offset_d_in,
+    EndOffsetIteratorInputT end_offset_d_in,
+    BeginOffsetIteratorOutputT begin_offset_d_out,
+    OffsetT n_segments,
+    ScanOpT scan_op,
+    InitValueT init_value)
+{
+  using segmented_scan_policy_t = typename ChainedPolicyT::ActivePolicy::SegmentedScanPolicyT;
+
+  using agent_segmented_scan_t = cub::detail::segmented_scan::AgentSegmentedScan<
+    segmented_scan_policy_t,
+    InputIteratorT,
+    OutputIteratorT,
+    OffsetT,
+    ScanOpT,
+    ActualInitValueT,
+    AccumT,
+    ForceInclusive>;
+
+  __shared__ typename agent_segmented_scan_t::TempStorage temp_storage;
+
+  const ActualInitValueT _init_value = init_value;
+
+  const auto segment_id = blockIdx.x;
+  if (segment_id < n_segments)
```

Comment:
Suggestion: make it an assertion so that this check is not compiled in release code